File: csharp/Ice/interceptor/ThermostatI.cs
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

using Demo;
using System;

class ThermostatI : ThermostatDisp_
{
    public override float getTemp(Ice.Current current)
    {
        lock(this)
        {
            return _temperature;
        }
    }

    public override void setTemp(float temp, Ice.Current current)
    {
        lock(this)
        {
            _temperature = temp;
        }
    }

    public override void shutdown(Ice.Current current)
    {
        Console.Out.WriteLine("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }

    // Temperature in Celsius.
    private float _temperature = 23.5f;
}

File: java/Ice/interceptor/ThermostatI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import com.zeroc.demos.Ice.interceptor.Demo.*;
import com.zeroc.Ice.Current;

class ThermostatI implements Thermostat
{
    @Override
    public synchronized float getTemp(Current current)
    {
        return _temperature;
    }

    @Override
    public synchronized void setTemp(float temp, Current current)
    {
        _temperature = temp;
    }

    @Override
    public void shutdown(Current current)
    {
        System.out.println("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }

    // Temperature in Celsius.
    private float _temperature = 23.5f;
}
